ql/ql/src/codeql_ql/performance/VarUnusedInDisjunctQuery.qll

```diff
@@ -0,0 +1,27 @@
+import ql
+
+/**
+ * Holds if we assume `t` is a small type, and
+ * variables of this type are therefore not an issue in cartesian products.
+ */
+predicate isSmallType(Type t) {
+  t.getName() = "string" // DataFlow::Configuration and the like
+  or
+  exists(NewType newType | newType = t.getDeclaration() |
+    forex(NewTypeBranch branch | branch = newType.getABranch() | branch.getArity() = 0)
```

Comment:
Here, and in the recursive case two disjuncts below, why'd you need `forex`, rather than `forall`?

Reply:
I know that a branch must exists (so `forall` and `forex` are equivalent).  
And `forex` tend to avoid some horrible join-orders that can sometimes appear with `forall`, so I usually go for `forex` when they are equivalent.

Reply:
👍